### ISSUE TITLE
Don't install unnecessary `StorageClass`es

### ISSUE THREAD

**Issue by guidoiaquinti** [PostHog/charts-clickhouse#174]
## Is your feature request related to a problem?

As part of the `clickhouse_instance.yaml` template definition, if we are installing the chart on GCP or AWS, we add (by default) an unnecessary `StorageClass` definition.

```
kubectl get storageclass

NAME            PROVISIONER             RECLAIMPOLICY   VOLUMEBINDINGMODE      ALLOWVOLUMEEXPANSION   AGE
gp2 (default)   kubernetes.io/aws-ebs   Delete          WaitForFirstConsumer   false                  15h
gp2-resizable   kubernetes.io/aws-ebs   Retain          Immediate              true                   15h
```

This `StorageClass` is then only referenced by `ClickHouseInstallation` and not by other stateful components. I think we should remove this unnecessary complication especially as it created confusions in the past.  A `StorageClass` is also a k8s global resource, we might hit naming collisions, etc… 

Unfortunately this will be a breaking change and it will require a safe migration strategy like:

1. make sure `helm` runs cleanly
1. make sure the default `StorageClass` has `VolumeExpansion` set to true
1. stop the ClickHouse pod (make sure we don't destroy the underlying PVC/PV)
1. use something like `pv-migrate` (see https://github.com/utkuozdemir/pv-migrate) to migrate the data from the source PV to the destination PV.
1. delete the `ClickHouseInstallation` definition
1. manually patch the `PersistentVolume` definition to use the new volumes
1. run `helm` upgrade with the new config to create a new `ClickHouseInstallation` definition

## Describe the solution you’d like
By default, PostHog shouldn't require any custom `StorageClass`.

## Describe alternatives you’ve considered
Do nothing.

**Comment by tiina303** [comment]
If we want to do this, it would be great if we could do this before getting new installs from product-hunt or (perhaps we can temporarily make new installs not have storage class and existing installs keep the storage class for now)?

**Comment by Legion2** [comment]
Also related to https://github.com/PostHog/charts-clickhouse/issues/215

**Comment by guidoiaquinti** [comment]
Closed via #224
